docs/docs/reference/contextual/givens.md

```diff
@@ -100,6 +100,24 @@ transparent inline given mkAnnotations[A, T]: Annotations[A, T] = ${
 
 Since `mkAnnotations` is `transparent`, the type of an application is the type of its right-hand side, which can be a proper subtype of the declared result type `Annotations[A, T]`.
 
+Given instances can have the `inline` but not `transparent` modifiers as their type is already known from the signature.
+Example:
+
+```scala
+trait Show[T] {
+  inline def show(x: T): String
+}
+
+inline given Show[Foo] with {
+  /*transparent*/ inline def show(x: Foo): String = ${ ... }
+}
```

Comment:
I could not come up with a useful example of inlining those givens that did not involve also inlining the method.

Reply:
Actually, the only reason why we did not disallow `inline given .. with ..` is for the case where it contains an inline method (as I originally intended in #14207).

Reply:
Could the doc describe a concrete situation where it's useful then? I imagine it's needed for the method call to actually be inlined but that should be shown explicitly with an example because it's not obvious

Reply:
> I imagine it's needed for the method call to actually be inlined but that should be shown explicitly with an example because it's not obvious

The use case of the issue at the origin of this PR was compile-time typeclass. The goal is to allow, as you said, to fully inline the method call. For example, I use it for type refinements/constraints.